### PR TITLE
host: add deault implementation for system.build.isoImage target (per…

### PR DESCRIPTION
**Pull request by blaggacao**
… host)

This is a more generic implementation to ISO builds, similar to what nixpkgs/flake.nix does with `lib.nixosSystem` for VMs.

In a follow up PR, I'll implement netboot and sd card flashing (in principle)
as prototyped here: https://github.com/numtide/flake-utils/pull/18/commits/d1ff2f04a010c79d4db26a81dc68662200e85862

---

Looks promising: 

```console
$ nix build .#nixosConfigurations.NixOS.config.system.build.iso
$ nix build .#nixosConfigurations.ci.config.system.build.iso
$ ls result/iso
nixos-NixOS-20.09pre-git-x86_64-linux.iso
# ... still waiting for results from ci build
# ... wait a minute: wouldn't this be a strategy for ci targets, too? (it looks like so) ... going to stop it
```

The wireless conflict (2nd commit) comes from the [installation device profile](https://github.com/NixOS/nixpkgs/blob/896a3428ecaea3a87ef456dded3cb8c67089b0dd/nixos/modules/profiles/installation-device.nix#L73)

---

EDIT: I fixed `flk iso NixOS` in latest force-push

### CODE REVIEW

**Comment by blaggacao** [comment]
preview idea for possible sd + netboot support: https://github.com/blaggacao/nixflk/pull/1

**Comment by nrdxp** [comment]
This looks great. Anything else you want to implement, or is it ready for merge?

**Comment by blaggacao** [comment]
This is ready for merge, going forward you might put a close eye on what would be necessary here: https://github.com/nrdxp/nixflk/pull/102/files#diff-9e2394d5d965447692e8b09d0235b8fdd8426071c2537d0e4a7e4786d9fcf97eR32-R33

Please also build on your machine to check if everything works (it worked over here).

I'm still fighting with sd-images for similar reasons on https://github.com/blaggacao/nixflk/pull/1 (while netboot seems good).

I might open another PR with netboot support, soonish though.

**Comment by nrdxp** [comment]
I'll give it a test drive and see if everything works out. If so I'll go ahead and merge.

**Review by nrdxp — CHANGES_REQUESTED** [review]
Basically just wondering if we can break out the logic in lib/default.nix instead of clutter the head of hosts/default.nix. Other than that, everything seems to work wonderfully.

File: hosts/default.nix [hunk 1]
```diff
@@ -15,8 +15,42 @@ let
   unstableModules = [ ];
   addToDisabledModules = [ ];
 
+  libExt = lib.extend (
```
[the remark this answers is not in the document]
Comment (blaggacao) [reply hosts/default.nix:18]: I think you are right, I just did educated, yet uncritical, copy pasting from `nixpkgs/flake.nix`

**Comment by blaggacao** [comment]
For future reference / inspiration: https://github.com/nix-community/nixos-generators, specifically: https://github.com/nix-community/nixos-generators/issues/85

**Review by nrdxp — CHANGES_REQUESTED** [review]
Oh, also please update the README.md to reflect the new changes.

**Comment by blaggacao** [comment]
Can you review again:

- Readme updated (force push)
- moved to `./lib`

**Review by nrdxp — CHANGES_REQUESTED** [review]
Sent a PR with some small style fixups to keep the line limit under control. Other than that, this looks ready to merge!

**Pull request merged**